src/types/Course/Preference.ts

```diff
@@ -70,3 +153,101 @@ export const PreferenceQuery = extendType({
     });
   },
 });
+
+export const PreferenceMutation = extendType({
+  type: 'Mutation',
+  definition(t) {
+    t.nonNull.field('createTeachingPreference', {
+      type: Response,
+      description: 'Teaching preferences',
+      args: {
+        input: arg({ type: nonNull(CreateTeachingPreferenceInput) }),
+      },
+      resolve: async (
+        _,
+        {
+          input: {
+            peng,
+            userId,
+            year,
+            CoursePreferenceInput,
+            hasRelief,
+            reliefReason,
+            hasTopic,
+            topicDescription,
+            maxFallCapacity,
+            maxSpringCapacity,
+            maxSummerCapacity,
+          },
+        },
+        { prisma }
+      ) => {
+        const courseObjs = await Promise.all(
+          CoursePreferenceInput.map(
+            async ({
+              subject,
+              code,
+              term,
+              preference,
+            }: {
+              subject: string;
+              code: string;
+              term: Term;
+              preference: number;
+            }) => {
+              const courseData = await (prisma as PrismaClient).course.findFirst({
+                where: {
+                  subject: subject ?? '',
+                  code: code ?? '',
+                  term: term ?? '',
+                  year: year ?? '',
+                },
```

Comment:
```suggestion
                where: {
                  subject,
                  code,
                  term,
                  year,
                },
```
nexus is able to handle empty strings, I took that out and it still builds fine for me, so it's a way of simplifying it

Reply:
Good catch, thanks.